core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -192,6 +208,167 @@ public void doCreateAdminUser(StaplerRequest req, StaplerResponse rsp) throws IO
         }
     }
 
+    /*package*/ static void completeUpgrade(Jenkins jenkins) throws IOException {
+        setCurrentLevel(Jenkins.getVersion());
+        // jenkins.getInstallState().proceedToNextState();
+    }
+    
+    /*package*/ static void setCurrentLevel(VersionNumber v) throws IOException {
+        FileUtils.writeStringToFile(getUpdateStateFile(), v.toString());
+    }
+    
+    /**
+     * File that captures the state of upgrade.
+     *
+     * This file records the version number that the installation has upgraded to.
+     */
+    /*package*/ static File getUpdateStateFile() {
+        return new File(Jenkins.getInstance().getRootDir(),"jenkins.install.UpgradeWizard.state");
+    }
+    
+    /**
+     * What is the version the upgrade wizard has run the last time and upgraded to?.
+     * If {@link #getUpdateStateFile()} is missing, presumes the baseline is 1.0
+     * @return Current baseline. {@code null} if it cannot be retrieved.
+     */
+    @Restricted(NoExternalUse.class)
+    @CheckForNull
+    public static VersionNumber getCurrentLevel() {
+        VersionNumber from = new VersionNumber("1.0");
+        File state = getUpdateStateFile();
+        if (state.exists()) {
+            try {
+                from = new VersionNumber(defaultIfBlank(readFileToString(state), "1.0").trim());
+            } catch (IOException ex) {
+                LOGGER.log(Level.SEVERE, "Cannot read the current version file", ex);
+                return null;
+            }
+        }
+        return from;
+    }
+    
+    /**
+     * Returns the initial plugin list in JSON format
+     */
+    @Restricted(DoNotUse.class) // WebOnly
+    public HttpResponse doPlatformPluginList() throws IOException {
+        jenkins.install.SetupWizard setupWizard = Jenkins.getInstance().getSetupWizard();
+        if (setupWizard != null) {
+            if (InstallState.UPGRADE.equals(Jenkins.getInstance().getInstallState())) {
+                JSONArray initialPluginData = getPlatformPluginUpdates();
+                if(initialPluginData != null) {
+                    return hudson.util.HttpResponses.okJSON(initialPluginData);
+                }
+            } else {
+                JSONArray initialPluginData = getPlatformPluginList();
+                if(initialPluginData != null) {
+                    return hudson.util.HttpResponses.okJSON(initialPluginData);
+                }
+            }
+        }
+        return hudson.util.HttpResponses.okJSON();
```

Comment:
Minor: there doesn't seem to be any reason for `HttpResponses` or `SetupWizard` to use fully qualified imports in this method.